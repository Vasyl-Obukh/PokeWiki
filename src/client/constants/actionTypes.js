export const CARDS_FETCHING = 'CARDS_FETCHING';
export const CARDS_RECEIVED = 'CARDS_RECEIVED';
export const CARDS_REQUESTED = 'CARDS_REQUESTED';
export const CARDS_ERROR = 'CARDS_ERROR';

export const PAGE_FETCHING = 'PAGE_FETCHING';
export const PAGE_RECEIVED = 'PAGE_RECEIVED';
export const PAGE_REQUESTED = 'PAGE_REQUESTED';
export const PAGE_ERROR = 'PAGE_ERROR';

export const RELATED_FETCHING = 'RELATED_FETCHING';
export const RELATED_RECEIVED = 'RELATED_RECEIVED';