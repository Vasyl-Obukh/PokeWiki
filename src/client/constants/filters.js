export default {
  types: [
    'normal',
    'fighting',
    'flying',
    'poison',
    'ground',
    'rock',
    'bug',
    'ghost',
    'steel',
    'fire',
    'water',
    'grass',
    'electric',
    'psychic',
    'ice',
    'dragon',
    'fairy'
  ],
  levels: [1, 2, 3]
};