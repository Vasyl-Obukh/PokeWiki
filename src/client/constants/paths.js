export default {
  HOME: '/',
  POKEMON_PAGE: '/pokemons/:id',
  ERROR: '/error'
};