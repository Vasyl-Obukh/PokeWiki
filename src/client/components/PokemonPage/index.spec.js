import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { PokemonPage, mapStateToProps, mapDispatchToProps } from './index';
import { PAGE_REQUESTED } from '../../constants/actionTypes';
import paths from '../../constants/paths';

describe('<PokemonPage /> tests', () => {
  const mockStore = configureStore();
  const initialState = {};
  const store = mockStore(initialState);
  let props;

  beforeEach(() => {
    props = {
      fetchPage: jest.fn(),
      data: {
        name: 'pikachu',
        types: ['electric'],
        abilities: ['lightning-rod', 'static'],
        stats: [{name: 'hp', base: 35}, {name: 'speed', base: 90}]
      },
      relatedData: [],
      isLoading: false,
      relatedIsLoading: false,
      error: undefined
    };
  });

  test('Should render spinner for main content', () => {
    props.isLoading = true;

    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <PokemonPage {...props}/>
        </Router>
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should render spinner for related content', () => {
    props.relatedIsLoading = true;

    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <PokemonPage {...props}/>
        </Router>
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should render error message block', () => {
    props.error = 'Something goes wrong...';

    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <PokemonPage {...props}/>
        </Router>
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should render related block', () => {
    props.relatedData = [
      {id: 1, name: 'bulbasaur'}
    ];

    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <PokemonPage {...props}/>
        </Router>
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should invoke fetchPage only once', () => {
    renderer.create(<PokemonPage {...props}/>);
    expect(props.fetchPage.mock.calls.length).toBe(1);
  });
});

describe('<PokemonPage /> connect functions tests', () => {
  let props, state;

  beforeEach(() => {
    state = {
      page: {data: {}, isLoading: false, error: undefined},
      related: {isLoading: false, data: []}
    };
    props = {
      location: {pathname: '/pokemon/34'}
    };
  });

  test('mapStateToProps() should work correctly', () => {
    expect(mapStateToProps(state, props)).toEqual({
      id: 34,
      data: {},
      isLoading: false,
      relatedIsLoading: false,
      relatedData: [],
      error: undefined
    });
  });

  test('mapDispatchToProps() should return correct action object', () => {
    const dispatch = jest.fn();
    const id = 7;

    mapDispatchToProps(dispatch).fetchPage(id);
    expect(dispatch.mock.calls[0][0]).toEqual({
      type: PAGE_REQUESTED,
      url: `${paths.API_ROOT}/pokemon/7`
    });
  });
});