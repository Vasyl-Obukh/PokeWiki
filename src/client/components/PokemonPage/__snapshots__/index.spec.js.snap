// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonPage /> tests Should render error message block 1`] = `
<div
  className="sc-VigVT iybjHG"
>
  <main
    className="sc-jTzLTM jTgrwB"
  >
    <h2
      className="sc-gqjmRU dTdyQT"
    >
      Something goes wrong...
    </h2>
    <div
      className="sc-bwzfXH gQGX"
    >
      <img
        alt="pikachu"
        className="sc-htpNat ihZMVi"
        src="#"
      />
      <div
        className="sc-bxivhb iMxPcQ"
      >
        <div
          className="sc-ifAKCX WMAKS"
        >
          Name
        </div>
        <div
          className="sc-EHOje uksKz"
        >
          pikachu
        </div>
        <div
          className="sc-ifAKCX WMAKS"
        >
          Abilities
        </div>
        <div
          className="sc-EHOje fgZDWy"
        >
          <span
            className="sc-bZQynM kXoohU"
          >
            lightning-rod
          </span>
          <span
            className="sc-bZQynM kXoohU"
          >
            static
          </span>
        </div>
        <div
          className="sc-ifAKCX WMAKS"
        >
          Types
        </div>
        <div
          className="sc-EHOje fgZDWy"
        >
          <span
            className="sc-bZQynM kXoohU"
          >
            electric
          </span>
        </div>
        <div
          className="sc-ifAKCX WMAKS"
        >
          Experience
        </div>
        <div
          className="sc-EHOje fgZDWy"
        />
        <div
          className="sc-ifAKCX WMAKS"
        >
          Height
        </div>
        <div
          className="sc-EHOje fgZDWy"
        />
        <div
          className="sc-ifAKCX WMAKS"
        >
          Weight
        </div>
        <div
          className="sc-EHOje fgZDWy"
        />
        <div
          className="sc-ifAKCX WMAKS"
        >
          hp
        </div>
        <div
          className="sc-EHOje fgZDWy"
        >
          35
        </div>
        <div
          className="sc-ifAKCX WMAKS"
        >
          speed
        </div>
        <div
          className="sc-EHOje fgZDWy"
        >
          90
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`<PokemonPage /> tests Should render related block 1`] = `
<div
  className="sc-VigVT iybjHG"
>
  <main
    className="sc-jTzLTM jTgrwB"
  >
    <div
      className="sc-bwzfXH gQGX"
    >
      <img
        alt="pikachu"
        className="sc-htpNat ihZMVi"
        src="#"
      />
      <div
        className="sc-bxivhb iMxPcQ"
      >
        <div
          className="sc-ifAKCX WMAKS"
        >
          Name
        </div>
        <div
          className="sc-EHOje uksKz"
        >
          pikachu
        </div>
        <div
          className="sc-ifAKCX WMAKS"
        >
          Abilities
        </div>
        <div
          className="sc-EHOje fgZDWy"
        >
          <span
            className="sc-bZQynM kXoohU"
          >
            lightning-rod
          </span>
          <span
            className="sc-bZQynM kXoohU"
          >
            static
          </span>
        </div>
        <div
          className="sc-ifAKCX WMAKS"
        >
          Types
        </div>
        <div
          className="sc-EHOje fgZDWy"
        >
          <span
            className="sc-bZQynM kXoohU"
          >
            electric
          </span>
        </div>
        <div
          className="sc-ifAKCX WMAKS"
        >
          Experience
        </div>
        <div
          className="sc-EHOje fgZDWy"
        />
        <div
          className="sc-ifAKCX WMAKS"
        >
          Height
        </div>
        <div
          className="sc-EHOje fgZDWy"
        />
        <div
          className="sc-ifAKCX WMAKS"
        >
          Weight
        </div>
        <div
          className="sc-EHOje fgZDWy"
        />
        <div
          className="sc-ifAKCX WMAKS"
        >
          hp
        </div>
        <div
          className="sc-EHOje fgZDWy"
        >
          35
        </div>
        <div
          className="sc-ifAKCX WMAKS"
        >
          speed
        </div>
        <div
          className="sc-EHOje fgZDWy"
        >
          90
        </div>
      </div>
    </div>
    <h3
      className="sc-htoDjs dLESzS"
    >
      Related pokemons
    </h3>
    <div
      className="sc-gzVnrw jXalhU"
    >
      <a
        className="sc-dnqmqq ceKmmb"
        href="/pokemon/1"
      >
        <img
          alt="bulbasaur"
          className="sc-iwsKbI kqtvOR"
        />
        <p
          className="sc-gZMcBi bImbSu"
        >
          bulbasaur
        </p>
      </a>
    </div>
  </main>
</div>
`;

exports[`<PokemonPage /> tests Should render spinner for main content 1`] = `
<div
  className="sc-VigVT iybjHG"
>
  <main
    className="sc-jTzLTM jTgrwB"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-spinner fa-w-16 sc-bdVaJa flmXWk"
      data-icon="spinner"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <div
      className="sc-bwzfXH gQGX"
    >
      <img
        alt="pikachu"
        className="sc-htpNat ihZMVi"
        src="#"
      />
      <div
        className="sc-bxivhb iMxPcQ"
      >
        <div
          className="sc-ifAKCX WMAKS"
        >
          Name
        </div>
        <div
          className="sc-EHOje uksKz"
        >
          pikachu
        </div>
        <div
          className="sc-ifAKCX WMAKS"
        >
          Abilities
        </div>
        <div
          className="sc-EHOje fgZDWy"
        >
          <span
            className="sc-bZQynM kXoohU"
          >
            lightning-rod
          </span>
          <span
            className="sc-bZQynM kXoohU"
          >
            static
          </span>
        </div>
        <div
          className="sc-ifAKCX WMAKS"
        >
          Types
        </div>
        <div
          className="sc-EHOje fgZDWy"
        >
          <span
            className="sc-bZQynM kXoohU"
          >
            electric
          </span>
        </div>
        <div
          className="sc-ifAKCX WMAKS"
        >
          Experience
        </div>
        <div
          className="sc-EHOje fgZDWy"
        />
        <div
          className="sc-ifAKCX WMAKS"
        >
          Height
        </div>
        <div
          className="sc-EHOje fgZDWy"
        />
        <div
          className="sc-ifAKCX WMAKS"
        >
          Weight
        </div>
        <div
          className="sc-EHOje fgZDWy"
        />
        <div
          className="sc-ifAKCX WMAKS"
        >
          hp
        </div>
        <div
          className="sc-EHOje fgZDWy"
        >
          35
        </div>
        <div
          className="sc-ifAKCX WMAKS"
        >
          speed
        </div>
        <div
          className="sc-EHOje fgZDWy"
        >
          90
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`<PokemonPage /> tests Should render spinner for related content 1`] = `
<div
  className="sc-VigVT iybjHG"
>
  <main
    className="sc-jTzLTM jTgrwB"
  >
    <div
      className="sc-bwzfXH gQGX"
    >
      <img
        alt="pikachu"
        className="sc-htpNat ihZMVi"
        src="#"
      />
      <div
        className="sc-bxivhb iMxPcQ"
      >
        <div
          className="sc-ifAKCX WMAKS"
        >
          Name
        </div>
        <div
          className="sc-EHOje uksKz"
        >
          pikachu
        </div>
        <div
          className="sc-ifAKCX WMAKS"
        >
          Abilities
        </div>
        <div
          className="sc-EHOje fgZDWy"
        >
          <span
            className="sc-bZQynM kXoohU"
          >
            lightning-rod
          </span>
          <span
            className="sc-bZQynM kXoohU"
          >
            static
          </span>
        </div>
        <div
          className="sc-ifAKCX WMAKS"
        >
          Types
        </div>
        <div
          className="sc-EHOje fgZDWy"
        >
          <span
            className="sc-bZQynM kXoohU"
          >
            electric
          </span>
        </div>
        <div
          className="sc-ifAKCX WMAKS"
        >
          Experience
        </div>
        <div
          className="sc-EHOje fgZDWy"
        />
        <div
          className="sc-ifAKCX WMAKS"
        >
          Height
        </div>
        <div
          className="sc-EHOje fgZDWy"
        />
        <div
          className="sc-ifAKCX WMAKS"
        >
          Weight
        </div>
        <div
          className="sc-EHOje fgZDWy"
        />
        <div
          className="sc-ifAKCX WMAKS"
        >
          hp
        </div>
        <div
          className="sc-EHOje fgZDWy"
        >
          35
        </div>
        <div
          className="sc-ifAKCX WMAKS"
        >
          speed
        </div>
        <div
          className="sc-EHOje fgZDWy"
        >
          90
        </div>
      </div>
    </div>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-spinner fa-w-16 sc-bdVaJa flmXWk"
      data-icon="spinner"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </main>
</div>
`;
